src/metrics/function_error.py

```diff
@@ -37,6 +37,28 @@ def evaluate(self, models: Iterable[ContextModel], num_batches: int = 1) -> Iter
 
         return errs
 
+class RegressionScore(Benchmark):
+    # zero_err is of shape (1)
+    # optimal_err is of shape (seq_length,)
+    def __init__(self, metric: Metric, function_class: FunctionClass, zero_err, optimal_err):
+        self.function_class = function_class
+        self.metric = metric
+        self.zero_err = zero_err
+        self.optimal_err = optimal_err.mean(dim = 1)
+
+    def evaluate(self, models: Iterable[ContextModel]) -> Iterable[Tensor]:
+        # generating model errors
+        model_funct = FunctionClassError(self.metric, self.function_class)
```

Comment:
This can be instantiated in `__init__`, all arguments are constant between invocations of `.evaluate`